canonicals.js

```diff
@@ -0,0 +1,53 @@
+var APP = APP || {};
+(function(app) {
+
+    var canonicals = {
+
+        closures: [750486],
+        asynchronous: [14220321],
+        inheritance: [11072556, 7486825]
+
+    };
+
+    var ids = Object.keys(canonicals).map(function(category) {
+        return canonicals[category].join(';');
+    });
+
+    app.xhr.fetch(ids.join(';')).then(function success(questions) {
+        questions.items.forEach(function(question) {
+            Object.keys(canonicals).forEach(function(category) {
+                canonicals[category].forEach(function(q, i) {
+                    //they see me nesting...
+                    if (q === question.question_id) {
+                        canonicals[category][i] = question;
```

Comment:
I wouldn't mutate the existing canonicals array (it kind of beats the point of using higher order functions like .forEach and .map), you have no guarantee for the order in which things run.

Creating a new Object with the resolved question objects is likely better.